### PR TITLE
[MGN-899] Added nullable fields to swap leg and swap and tests for se…

### PR DESCRIPTION
**Pull request by traggatt**
…rialization and deserialization

### CODE REVIEW

File: projects/OG-FinancialTypes/src/main/java/com/opengamma/financial/convention/FixedInterestRateSwapLegConvention.java [hunk 1]
```diff
@@ -149,14 +150,16 @@ public ConventionType getConventionType() {
    * @param rate  the interest rate (may be simple or complex)
```
Comment (jonathansenior) [inline projects/OG-FinancialTypes/src/main/java/com/opengamma/financial/convention/FixedInterestRateSwapLegConvention.java]: Javadoc for the new parameters

File: projects/OG-FinancialTypes/src/main/java/com/opengamma/financial/convention/FloatingInterestRateSwapLegConvention.java [hunk 2]
```diff
@@ -282,13 +283,15 @@ public ConventionType getConventionType() {
    * @param payOrReceive  is this a pay or receive leg?
    * @return the leg, not null
    */
-  public FloatingInterestRateSwapLeg toLeg(final InterestRateSwapNotional notional, final PayReceiveType payOrReceive) {
+  public FloatingInterestRateSwapLeg toLeg(final InterestRateSwapNotional notional, final PayReceiveType payOrReceive, final LocalDate effectiveDate, final LocalDate terminationDate) {
```
Comment (jonathansenior) [inline projects/OG-FinancialTypes/src/main/java/com/opengamma/financial/convention/FloatingInterestRateSwapLegConvention.java]: Javadoc for the new parameters

File: projects/OG-FinancialTypes/src/main/java/com/opengamma/financial/convention/FixedInterestRateSwapLegConvention.java [hunk 3]
```diff
@@ -149,14 +150,16 @@ public ConventionType getConventionType() {
...
    * @return the leg, not null
    */
-  public FixedInterestRateSwapLeg toLeg(final InterestRateSwapNotional notional, final PayReceiveType payOrReceive, final Rate rate) {
+  public FixedInterestRateSwapLeg toLeg(final InterestRateSwapNotional notional, final PayReceiveType payOrReceive, final Rate rate, final LocalDate effectiveDate, final LocalDate terminationDate) {
```
Comment (jonathansenior) [inline projects/OG-FinancialTypes/src/main/java/com/opengamma/financial/convention/FixedInterestRateSwapLegConvention.java]: Javadoc for the new parameters

File: projects/OG-FinancialTypes/src/main/java/com/opengamma/financial/security/irs/InterestRateSwapSecurity.java [hunk 4]
```diff
@@ -121,6 +147,52 @@ public InterestRateSwapLeg getReceiveLeg() {
...
+      } else {
+        return leg2;
+      }
+    }
+  }
+
+  /**
+   * If termination date is present at the swap level, that is the value to use.
+   * If the termination date is null, then there must be termination dates defined on the legs.
+   * In this case return the minimum value between the two legs.
```
Comment (jonathansenior) [inline projects/OG-FinancialTypes/src/main/java/com/opengamma/financial/security/irs/InterestRateSwapSecurity.java]: Typo here - think you mean maximum

File: projects/OG-FinancialTypes/src/test/java/com/opengamma/financial/security/irs/InterestRateSwapLegSerializationTest.java [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+    InputStream is = InterestRateSwapLegSerializationTest.class.getClassLoader().getResourceAsStream(legacySwapLegWithNoStartDateOrEndDatePath);
+    FloatingInterestRateSwapLeg swapLeg = JodaBeanSerialization.deserializer().xmlReader().read(is, FloatingInterestRateSwapLeg.class);
+    Assert.assertEquals(swapLeg.getNotional().getInitialAmount(), 1000000D);
+    Assert.assertNotNull(swapLeg.getEffectiveDate());
+    Assert.assertNotNull(swapLeg.getUnadjustedMaturityDate());
+    Assert.assertEquals(swapLeg.getEffectiveDate(), LocalDate.of(2014, 2, 1));
+    Assert.assertEquals(swapLeg.getUnadjustedMaturityDate(), LocalDate.of(2024, 4, 2));
+  }
+
+//  @Test
```
Comment (jonathansenior) [inline projects/OG-FinancialTypes/src/test/java/com/opengamma/financial/security/irs/InterestRateSwapLegSerializationTest.java]: Understand why this is here but should probably remove it.

File: projects/OG-FinancialTypes/src/main/java/com/opengamma/financial/convention/FloatingInterestRateSwapLegConvention.java (hunk 2 above)
Comment (jacbop) [inline projects/OG-FinancialTypes/src/main/java/com/opengamma/financial/convention/FloatingInterestRateSwapLegConvention.java]: fixed

File: projects/OG-FinancialTypes/src/main/java/com/opengamma/financial/convention/FixedInterestRateSwapLegConvention.java (hunk 3 above)
Comment (jacbop) [inline projects/OG-FinancialTypes/src/main/java/com/opengamma/financial/convention/FixedInterestRateSwapLegConvention.java]: fixed

File: projects/OG-FinancialTypes/src/main/java/com/opengamma/financial/security/irs/InterestRateSwapSecurity.java (hunk 4 above)
Comment (jacbop) [inline projects/OG-FinancialTypes/src/main/java/com/opengamma/financial/security/irs/InterestRateSwapSecurity.java]: yep, copy and paste error. fixed.

File: projects/OG-FinancialTypes/src/test/java/com/opengamma/financial/security/irs/InterestRateSwapLegSerializationTest.java (hunk 5 above)
Comment (jacbop) [inline projects/OG-FinancialTypes/src/test/java/com/opengamma/financial/security/irs/InterestRateSwapLegSerializationTest.java]: fixed.

**Comment by jacbop** [comment]
retest this please

**Comment by jonathansenior** [comment]
retest this please

**Comment by jacbop** [comment]
retest this please

**Pull request merged**